keyboards/cipulot/common/ec_switch_matrix.c

```diff
@@ -0,0 +1,312 @@
+ec_config_t        ec_config;
+
+// Pin and port array
+const pin_t row_pins[]                                 = MATRIX_ROW_PINS;
+const pin_t amux_sel_pins[]                            = AMUX_SEL_PINS;
+const pin_t amux_en_pins[]                             = AMUX_EN_PINS;
+const pin_t amux_n_col_sizes[]                         = AMUX_COL_CHANNELS_SIZES;
+const pin_t amux_n_col_channels[][AMUX_MAX_COLS_COUNT] = {AMUX_COL_CHANNELS};
+
+#define AMUX_SEL_PINS_COUNT ARRAY_SIZE(amux_sel_pins)
+#define EXPECTED_AMUX_SEL_PINS_COUNT ceil(log2(AMUX_MAX_COLS_COUNT)
+// Checks for the correctness of the configuration
+_Static_assert(ARRAY_SIZE(amux_en_pins) == AMUX_COUNT, "AMUX_EN_PINS doesn't have the minimum number of bits required to enable all the multiplexers available");
+// Check that number of select pins is enough to select all the channels
+_Static_assert(AMUX_SEL_PINS_COUNT == EXPECTED_AMUX_SEL_PINS_COUNT), "AMUX_SEL_PINS doesn't have the minimum number of bits required address all the channels");
+// Check that number of elements in AMUX_COL_CHANNELS_SIZES is enough to specify the number of channels for all the multiplexers available
+_Static_assert(ARRAY_SIZE(amux_n_col_sizes) == AMUX_COUNT, "AMUX_COL_CHANNELS_SIZES doesn't have the minimum number of elements required to specify the number of channels for all the multiplexers available");
+
+static uint16_t sw_value[MATRIX_ROWS][MATRIX_COLS];
+
+static adc_mux adcMux;
+
+// Initialize the row pins
+void init_row(void) {
+    // Set all row pins as output and low
+    for (uint8_t idx = 0; idx < MATRIX_ROWS; idx++) {
+        setPinOutput(row_pins[idx]);
+        writePinLow(row_pins[idx]);
+    }
+}
+
+// Initialize the multiplexers
+void init_amux(void) {
+    for (uint8_t idx = 0; idx < AMUX_COUNT; idx++) {
+        setPinOutput(amux_en_pins[idx]);
+        writePinLow(amux_en_pins[idx]);
+    }
+    for (uint8_t idx = 0; idx < AMUX_SEL_PINS_COUNT; idx++) {
+        setPinOutput(amux_sel_pins[idx]);
+    }
+}
+
+// Select the multiplexer channel of the specified multiplexer
+void select_amux_channel(uint8_t channel, uint8_t col) {
+    // Get the channel for the specified multiplexer
+    uint8_t ch = amux_n_col_channels[channel][col];
+    // momentarily disable specified multiplexer
+    writePinHigh(amux_en_pins[channel]);
+    // Select the multiplexer channel
+    for (uint8_t i = 0; i < AMUX_SEL_PINS_COUNT; i++) {
+        writePin(amux_sel_pins[i], ch & (1 << i));
+    }
+    // re enable specified multiplexer
+    writePinLow(amux_en_pins[channel]);
+}
+
+// Disable all the unused multiplexers
+void disable_unused_amux(uint8_t channel) {
+    // disable all the other multiplexers apart from the current selected one
+    for (uint8_t idx = 0; idx < AMUX_COUNT; idx++) {
+        if (idx != channel) {
+            writePinHigh(amux_en_pins[idx]);
+        }
+    }
+}
+// Discharge the peak hold capacitor
+void discharge_capacitor(void) {
+#if OPEN_DRAIN_SUPPORT
```

Comment:
Shouldn't this, and the others be:
```suggestion
#ifdef OPEN_DRAIN_SUPPORT
```

Reply:
Good catch, it should be `#ifdef` indeed. Fixed and pushed in the last commit.